### DISCUSSION TITLE
"Object reference not set to an instance of an object"

### DISCUSSION THREAD

**Discussion by nilssohn** [GSDragoon/RadeonSoftwareSlimmer#37] [General]
Hi and thank you for this tool.

The message above appears after installing a Radeon driver slimmed with RSS and then running your program again. This is happening not only with the current version 1.10.0 .NET 7, but also before in 1.8.0 .NET 6 and earlier. It is also not dependent on a specific Radeon driver version. I am noticing this for several months now. O/S is Win 11 pro 22H2. 

Today it happended with 23.5.2 after having slimmed the driver installation at its maximum in "Pre Install" ("Select None" everywhere except "AMD Display Driver").

![RSS_Object_reference](https://github.com/GSDragoon/RadeonSoftwareSlimmer/assets/136918186/f4cb2e9a-45af-47b9-9531-e2e53b815ef3)

Please tell me if more information is needed to fix this. Thank you.

**Comment by nilssohn** [comment]
@GSDragoon: I noticed that somebody (you?) put a "bug"-label to my posting above. Does this mean you will look into it? Any confirmation would be appreciated. If you need further testing or additional information, please don't hesitate to contact me.

**Reply by GSDragoon** [reply]
Yes, that was me. The logging and information about your setup was helpful and I believe to know where the issue is. I'm hoping to be able work on this some more this weekend and get a fixed version out.

**Comment by GSDragoon** [comment]
I put out a new release with what I believe should fix the issue you are running into. It should disable the host services section and allow you to use the other Post Install sections. Let me know how it goes.

https://github.com/GSDragoon/RadeonSoftwareSlimmer/releases/tag/1.10.1

**Comment by nilssohn** [comment]
Thank you, this is working. No more error.

>  It should disable the host services section and allow you to use the other Post Install sections.

Confirmed. Is this a functionality limitation compared to 1.10.0?

![Screenshot 2023-06-25 215523](https://github.com/GSDragoon/RadeonSoftwareSlimmer/assets/136918186/0ee5d363-b69f-429a-8c11-9ad7a329ac61)

**Reply by GSDragoon** [reply]
If you aren't installing all the components, yes. The main Radeon Software host service (the main Radeon Software) application is only available if you install the Radeon Software Settings component. The others are available if you install AMD DVR (provides overlay, recording, etc functionality). In order to gracefully stop them, I use their own CLI. You only get the start/restart and stop functionality if that CLI application exists. If you really are doing only the driver install, then none of that impacts you at all, thus it's empty and greyed out.

The other sections for Post Install (scheduled tasks, system services installations and temp files) should work now. That was the bug, the entirety of post install was failing instead of just disabling the part that doesn't apply.

**Comment by nilssohn** [comment]
Thank you. I confirm: with these Pre Install settings ...

![Screenshot 2023-06-26 100344](https://github.com/GSDragoon/RadeonSoftwareSlimmer/assets/136918186/dda3065a-dace-4943-ba89-3bf0213a51ce)

... I get these Post Install Host Service options.

![Screenshot 2023-06-26 101029](https://github.com/GSDragoon/RadeonSoftwareSlimmer/assets/136918186/692f3a00-a2ac-4f0f-b1c8-d82da2d9dccc)

So, you consider the bug fixed and 1.10.1 the current official version?

**Reply by GSDragoon** [reply]
> So, you consider the bug fixed and 1.10.1 the current official version?

Yes

**Comment by nilssohn** [comment]
Thank you.

Please let me use our discussion to ask one slightly offtopic question: Is there a "switch" in Radeon Software Slimmer to prevent AMD Noise Suppression feature to be active in the installation and in Windows Autostart?
